doc: add known issue for the upgrade (#427)
harvester/docs: docs/upgrade/v1-1-2-to-v1-2-0.md

  - [[BUG] 3 Node AirGapped Cluster Upgrade Stuck v1.1.0->v1.1.2-rc4](https://github.com/harvester/harvester/issues/3730 )

#### 5.3 The `instance-manager-e` pod could not be drained

In some scenarios, during an upgrade, you might encounter an issue where the `instance-manager-e` pod cannot be drained. This situation is typically indicated by error messages in the Rancher logs, like the ones shown below:

```
$ kubectl logs deployment/rancher -n cattle-system | grep "evicting pod"
evicting pod longhorn-system/instance-manager-r-a06a43f3437ab4f643eea7053b915a80
evicting pod longhorn-system/instance-manager-e-452e87d2
error when evicting pods/"instance-manager-r-a06a43f3437ab4f643eea7053b915a80" -n "Longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
error when evicting pods/"instance-manager-e-452e87d2" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
```

Then, you need to check the `instance-manager-e` to see if any engine instances remain.

```
$ kubectl get instancemanager instance-manager-e-452e87d2 -n longhorn-system -o yaml | yq -e ".status.instances"

> pvc-7b120d60-1577-4716-be5a-62348271025a-e-1cd53c57

Reviewer: The trailing colon of the engine name seems to be missing in the first line of the command output.

```suggestion
pvc-7b120d60-1577-4716-be5a-62348271025a-e-1cd53c57:
```

  spec:
    name: pvc-7b120d60-1577-4716-be5a-62348271025a-e-1cd53c57
  status:
    endpoint: ""
    errorMsg: ""
    listen: ""
    portEnd: 10001
    portStart: 10001
    resourceVersion: 0
    state: running
    type: ""
```

That means the `instance-manager-e-452e87d2` still has an engine instance so it can't be drained.

You need to check the engine number to see if anyone is redundant. Each PVC should only have one engine.

```

> # kubectl get engines -A | grep pvc-7b120d60-1577-4716-be5a-62348271025a

Reviewer: Maybe we can use `kubectl get engines -n longhorn-system -l longhornvolume=pvc-7b120d60-1577-4716-be5a-62348271025a` instead to promote the usage of the label. What do you think?

Author: Sure, nice suggestion!

- Related issue:
  - [[BUG] 3 Node AirGapped Cluster Upgrade Stuck v1.1.0->v1.1.2-rc4](https://github.com/harvester/harvester/issues/3730 )

#### 5.3 The `instance-manager-e` pod could not be drained

> If the `instance-manager-e` pod could not be drained, you might see the following error message in the Rancher logs:

Reviewer 2: ```suggestion
In some scenarios, during an upgrade, you might encounter an issue where the `instance-manager-e` pod cannot be drained. This situation is typically indicated by error messages in the Rancher logs, like the ones shown below:
```

> In some scenarios, during an upgrade, you might encounter an issue where the `instance-manager-e` pod cannot be drained. This situation is typically indicated by error messages in the Rancher logs, like the ones shown below:

Reviewer 3: ```suggestion
During an upgrade, you might encounter an issue where you can't drain the `instance-manager-e` pod. When this situation occurs, you will see error messages in the Rancher logs like the ones shown below:
```

```
$ kubectl logs deployment/rancher -n cattle-system | grep "evicting pod"
evicting pod longhorn-system/instance-manager-r-a06a43f3437ab4f643eea7053b915a80
evicting pod longhorn-system/instance-manager-e-452e87d2
error when evicting pods/"instance-manager-r-a06a43f3437ab4f643eea7053b915a80" -n "Longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
error when evicting pods/"instance-manager-e-452e87d2" -n "longhorn-system" (will retry after 5s): Cannot evict pod as it would violate the pod's disruption budget.
```

> Then, you need to check the `instance-manager-e` to see if it remains any engine.

Reviewer 2: ```suggestion
Then, you need to check the `instance-manager-e` to see if any engine instances remain.
```

> Then, you need to check the `instance-manager-e` to see if any engine instances remain.

Reviewer 3: ```suggestion
Check the `instance-manager-e` to see if any engine instances remain.
```

```
$ kubectl get instancemanager instance-manager-e-452e87d2 -n longhorn-system -o yaml | yq -e ".status.instances"
pvc-7b120d60-1577-4716-be5a-62348271025a-e-1cd53c57:
  spec:
    name: pvc-7b120d60-1577-4716-be5a-62348271025a-e-1cd53c57
  status:
    endpoint: ""
    errorMsg: ""
    listen: ""
    portEnd: 10001
    portStart: 10001
    resourceVersion: 0
    state: running
    type: ""
```

> That means the `instance-manager-e-452e87d2` still has an engine instance so it can't be drained.

Reviewer 3: ```suggestion
In this example, the `instance-manager-e-452e87d2` still has an engine instance, so you can't drain the pod.
```

You need to check the engine number to see if anyone is redundant. Each PVC should only have one engine.

```

> # kubectl get engines -A |grep pvc-7b120d60-1577-4716-be5a-62348271025a

Reviewer 4: ```suggestion
# kubectl get engines -A | grep pvc-7b120d60-1577-4716-be5a-62348271025a
```

> You need to check the engine number to see if anyone is redundant. Each PVC should only have one engine.

Reviewer 3: ```suggestion
You need to check the engine numbers to see if any engine number is redundant. Each PVC should only have one engine.
```

longhorn-system pvc-76120d60-1577-4716-be5a-62348271025a-e-08220662 running harvester-qv4hd instance-manager-e-625d715e2f2e7065d64339f9b31407c2 longhornio/longhorn-engine:v1.4.3  2d12h
longhorn-system pvc-7b120d60-1577-4716-be5a-62348271025a-e-lcd53c57 running harvester-lhlkv instance-manager-e-452e87d2                         longhornio/longhorn-engine:v1.4.3  4d10h
```

> You can confirm we have two engines for the same PVC. There is a known issue and we are working on it.
> You can delete the redundant engine to fix this issue and the upgrade will continue.
> 
> You can find the correct engine by following command:

Reviewer 2: ```suggestion
In the example above, you can see that two engines exist for the same PVC, which is a known issue in Longhorn [#6642](https://github.com/longhorn/longhorn/issues/6642). To resolve this, delete the redundant engine to allow the upgrade to continue. 

To determine which engine is the correct one, use the following command:
```

> In the example above, you can see that two engines exist for the same PVC, which is a known issue in Longhorn [#6642](https://github.com/longhorn/longhorn/issues/6642). To resolve this, delete the redundant engine to allow the upgrade to continue. 

Reviewer 3: ```suggestion
The example above shows that two engines exist for the same PVC, which is a known issue in Longhorn [#6642](https://github.com/longhorn/longhorn/issues/6642). To resolve this, delete the redundant engine to allow the upgrade to continue. 
```

```
$ kubectl get volumes pvc-7b120d60-1577-4716-be5a-62348271025a -n longhorn-system
NAME                                      STATE     ROBUSTNESS  SCHEDULED SIZE        NODE            AGE
pvc-7b120d60-1577-4716-be5a-62348271025a  attached  healthy               42949672960 harvester-q4vhd 4d10h
```

> The volume `pvc-7b120d60-1577-4716-be5a-62348271025a` is active on the node `harvester-q4vhd`, so the engine not running on this node is redundant.
> 
> The `pvc-7b120d60-1577-4716-be5a-62348271025a-e-lcd53c57` is redundant by the above command.
> 
> Try to make this engine inactive. Then, it would be deleted by Longhorn automatically.

Reviewer 2: ```suggestion
In this example, the volume `pvc-7b120d60-1577-4716-be5a-62348271025a` is active on the node `harvester-q4vhd`, indicating that the engine not running on this node is redundant.

To make the engine inactive and trigger its automatic deletion by Longhorn, run the following command:

```

```
$ kubectl patch engine pvc-7b120d60-1577-4716-be5a-62348271025a-e-lcd53c57 -n longhorn-system --type='json' -p='[{"op": "replace", "path": "/spec/active", "value": false}]'
engine.longhorn.io/pvc-7b120d60-1577-4716-be5a-62348271025a-e-lcd53c57 patched
```

> You can recheck the engine status a few seconds later.

Reviewer 2: ```suggestion
After a few seconds, you can verify the engine's status:
```

> 
> 

Reviewer 3: ```suggestion
```

```
$ kubectl get engine -n longhorn-system|grep pvc-7b120d60-1577-4716-be5a-62348271025a
pvc-7b120d60-1577-4716-be5a-62348271025a-e-08220b62   running  harvester-q4vhd   instance-manager-e-625d715e2f2e7065d64339f9631407c2  longhornio/longhorn-engine:v1.4.3   2d13h
```

> Now, the instance-manager-e should be drained successfully and the upgrade will continue.

Reviewer 2: ```suggestion
Now, the `instance-manager-e` should be successfully drained, allowing the upgrade to proceed.
```

> Now, the `instance-manager-e` should be successfully drained, allowing the upgrade to proceed.

Reviewer 3: ```suggestion
The `instance-manager-e` pod should now drain successfully, allowing the upgrade to proceed.
```
